handler/handler_test.go

```diff
@@ -83,9 +95,31 @@ func TestDataImporterHandle(t *testing.T) {
 	Convey("Given a handler configured with sucessful es updates", t, func() {
 		esDestURL := "locahost:9999"
 
-		//ES Client initialisation : this needs to be mock
 		doFuncWithInValidResponse := func(ctx context.Context, req *http.Request) (*http.Response, error) {
```

Comment:
We can extend and correct the tests so the first request using the Elasticsearch client returns a particular response and then a second request returns a different response.

You can manipulate the responses in test by doing something like:

```
		var count int
		doFuncWithInValidResponse := func(ctx context.Context, req *http.Request) (*http.Response, error) {
			count++
			if count == 1 {
				return successWithESResponseError(), nil
			} else {
				return successWithESResponseNoError(), nil
			}
		}
```

In the above we can verify the number of calls to mocked elasticsearch client by checking the count variable:

`So(count, ShouldEqual, 1)` or `So(count, ShouldEqual, 2)` depending on scenario, see scenarios below:

Test scenarios:

1) success with no failed resources
  a) Create bulk request succeeded with no failed resources
  b) Update bulk request not made
2) success with 1 or more failed resources followed by success with no failed resources
  a) Create bulk request succeeded with failed resources
  b) Update bulk request succeeded with no failed resources
3) success with 1 or more failed resources followed by success with 1 or more failed resources
  a) Create bulk request succeeded with failed resources
  b) Update bulk request succeeded with failed resources
4) failed to create resources (internal server error)
  b) Create bulk request failed
  b) Update bulk request not made
5) success with 1 or more failed resources followed by failed request (internal server error)
  b) Create bulk request succeeded with failed resources
  b) Update bulk request failed

Reply:
All the requested scenarios now added